src/fromager/overrides.py

```diff
@@ -83,6 +83,18 @@ def patches_for_source_dir(
     return sorted((patches_dir / source_dir_name).glob("*.patch"))
 
 
+def patches_for_req(
+    patches_dir: pathlib.Path, req: str
+) -> typing.Iterable[pathlib.Path]:
```

Comment:
I think a better return type would be a `list` since we are using `len` in the warn function and `len` is not necessarily part of an iterable. @tiran might be able to confirm?

Reply:
Okay, I have set the return type to list now. I will change it again if required.